src/main/java/core/basesyntax/strategy/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,15 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.Transaction;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    @Override
+    public void apply(Transaction transaction) {
+        Fruit fruit = transaction.getFruit();
+        Integer currentQuantity = Storage.get(fruit);
+        Storage.put(fruit, (currentQuantity == null ? 0
+                : currentQuantity) - transaction.getQuantity());
```

Comment:
maybe let's throw an exception if there is no such fruit or its quantity is not enough?